make/stanc

```diff
@@ -24,24 +24,40 @@ else
 endif
 
 ifeq ($(CMDSTAN_SUBMODULES),1)
-ifeq ($(STANC3),)
-bin/stanc$(EXE) :
+# bin/stanc build rules
+ifneq (,$(STANC2))
+bin/stanc$(EXE) : O = $(O_STANC)
+bin/stanc$(EXE) : CPPFLAGS_MPI =
+bin/stanc$(EXE) : LDFLAGS_MPI =
+bin/stanc$(EXE) : LDLIBS_MPI =
+bin/stanc$(EXE) : bin/cmdstan/libstanc.a
+bin/stanc$(EXE) : bin/cmdstan/stanc.o
 	@mkdir -p $(dir $@)
-ifeq ($(OS_TAG),windows)
-	$(shell echo "curl -L https://github.com/stan-dev/stanc3/releases/download/nightly/$(OS_TAG)-stanc -o bin/stanc$(EXE)")
-else
-	curl -L https://github.com/stan-dev/stanc3/releases/download/nightly/$(OS_TAG)-stanc -o bin/stanc
-	chmod +x bin/stanc
-endif
-else
+	$(LINK.cpp) $^ $(LDLIBS) $(OUTPUT_OPTION)
+
+else ifneq ($(STANC3),)
 bin/stanc$(EXE) : $(shell find $(STANC3)/src/ -type f -name '*.ml*') $(STANC#)
 	@mkdir -p $(dir $@)
 	cd $(STANC3) && echo "--- Rebuilding stanc ---\n" && dune build @install
 	cp $(STANC3)/_build/default/src/stanc/stanc.exe $@
+
+else ifneq (,$(wildcard bin/$(OS_TAG)-stanc))
+bin/stanc$(EXE) :
+	cp bin/$(OS_TAG)-stanc) bin/stanc$(EXE)
+	chmod +x bin/stanc
+
+else ifeq ($(OS_TAG),windows)
+bin/stanc$(EXE) :
+	@mkdir -p $(dir $@)
+https://github.com/stan-dev/stan/releases/latest
+	$(shell echo "curl -L https://github.com/stan-dev/stanc3/releases/download/latest/$(OS_TAG)-stanc -o bin/stanc$(EXE)")
```

Comment:
Do we want to use 

a) https://github.com/stan-dev/stanc3/releases/download/latest/$(OS_TAG)-stanc 
or should we rather go with 
b) https://github.com/stan-dev/stanc3/releases/download/v2.22.0/$(OS_TAG)-stanc

It probably does not make a huge difference either way since the release tarballs will ship with bin/linux-stanc, bin/windows-stanc, bin/mac-stanc binaries and this wont get called if someone uses the tarballs. Except if they somehow delete those files in which case a) would install the newest release and we could have a situation of someone using cmdstan 2.22 with stanc binary for version 2.23. But if someone does go and delete those files, they are playing with fire anyways, so not worth using b) just for this case?

Going with b) also means that develop will then use 2.22 tagged parser binary and we have to make a PR to change the version number each time.

Reply:
let's go with option b - when people are working from a release, they should get the fixed, tagged quantity.  if they want to get the nightly builds, they can work from dev.

Reply:
let's go with option a) - per comments in PR